ui/app/components/file-to-array-buffer.js

```diff
@@ -0,0 +1,57 @@
+import Component from '@ember/component';
+import filesize from 'filesize';
+
+export default Component.extend({
+  classNames: ['box', 'is-fullwidth', 'is-marginless', 'is-shadowless'],
+  onChange: () => {},
+  file: null,
+  fileName: null,
+  fileSize: null,
+  fileLastModified: null,
+
+  /*
+   * @public
+   * @param String
+   * Text to use as the label for the file input
+   * If null, a default will be rendered
+   */
+  label: null,
+
+  /*
+   * @public
+   * @param String
+   * Text to use as help under the file input
+   * If null, a default will be rendered
+   */
+  fileHelpText: null,
+
+  readFile(file) {
+    const reader = new FileReader();
+    reader.onload = () => this.send('onChange', reader.result, file);
+    reader.readAsArrayBuffer(file);
```

Comment:
do we need to explicitly return anything here?

Reply:
Nope! The `reader.onload` will fire when `readAsArrayBuffer` completes.